proposals/lambda-method-group-defaults.md

```diff
@@ -208,33 +283,80 @@ e = f; // Not Allowed
 b = f; // Allowed
 e = g; // Allowed
 
-d = (int c = 10) => 2; // Error: default parameter is different between new lambda
+d = (int c = 10) => 2; // Error: default parameter value is different between new lambda
                        // and synthesized delegate b'. We won't do implicit conversion
 ```
 
-Similarly, there is of course compatibility with named delegates that have default parameters as well: 
+Anonymous delegates with an array as the last parameter will be unified when the last parameter has the same `params` modifier, regardless of its name.
+
+```csharp
+int C(int[] xs) {
+  return xs.Length;
+}
+
+int D(params int[] xs) {
+  return xs.Length;
+}
+
+var a = (int[] xs) => xs.Length;
+// internal delegate int a'(int[] xs);
+var b = (params int[] xs) => xs.Length;
+// internal delegate int b'(params int[] xs);
+
+var c = C;
+// internal delegate int a'(int[] xs);
+var d = D;
+// internal delegate int b'(params int[] xs);
+
+a = b; // Not allowed
+a = c; // Allowed
+b = c; // Not allowed
+b = d; // Allowed
+
+c = (params int[] xs) => xs.Length; // Error: different delegate types; no implicit conversion
+d = (int[] xs) => xs.Length; // Error: different delegate types; no implicit conversion
+```
+
+Similarly, there is of course compatibility with named delegates that already support optional and `params` parameters.
 ```csharp
 int D(int a = 1) {
   return a;
 }
 
 delegate int Del(int a = 1);
 
-Del del = (int x = 100) => x; // Error: Default parameter does not match, so no conversion can be performed
+Del del = (int x = 100) => x; // Error: Default parameter value does not match, so no conversion can be performed
                               
 Del del1 = (int x = 1) => x; // Allowed, because default parameter value in lambda matches default parameter value in delegate
 
 Del del2 = D; // This behavior does not change and compiles as before as per the method group conversion rules 
 
 var d = D;
-// d is inferred as internal delegate int a'(int arg0 = 1);
+// d is inferred as internal delegate int a'(int arg = 1);
 
 Del del3 = d; // Not allowed. Cannot convert internal delegate type to Del.
               // Note that there is no change here from previous behavior, when d would be inferred
               // to be Action<int> since Action<int> also cannot be converted to a named delegate type.
+
+int E(params int[] xs) {
+  return xs.Length;
+}
+
+delegate int DelParams(params int[] xs);
+
+DelParams del4 = (int[] xs) => xs.Length; // Error: `params` modifier does not match, so no conversion can be performed
+
+DelParams del5 = (params int[] xs) => xs.Length; // Allowed
+
+DelParams del6 = E; // Allowed as before per the method group conversion rules
+
+var e = E;
+// e is inferred as internal delegate int b'(params int[] arg);
+
+DelParams del7 = e; // Not allowed. Cannot convert internal delegate type to DelParams.
 ```
 
```

Comment:
Consider including a conversion to non-`params` delegate if not already covered:
```csharp
delegate int DelArray(int[] xs);
DelArray d8 = (params int[] xs) => xs.Length; // Allowed
d8 = E; // Allowed
var e = (params int[] xs) => xs.Length;
d8 = e; // Not allowed. Cannot convert internal delegate type to DelArray.
```

Reply:
Should an error be reported converting a `params` lambda expression to a non-`params` delegate?

Reply:
> Consider including a conversion to non-`params` delegate

Good catch, thanks. Included: 728d5b5a8a5ab34e4a0f7dfd100c0faefe1fc107

> Should an error be reported converting a `params` lambda expression to a non-`params` delegate?

I believe yes for similar reason as converting a lambda expression with optional parameter to a delegate without: the default value / `params` modifier would effectively useless.